### PR TITLE
Fix arkode 3 and 4

### PR DESCRIPTION
**Pull request by drwells**
Part of #13703.

I tested this with both SUNDIALS 5.4 and SUNDIALS 6.2. Like the other test fixes, the right fix is to just lower the numerical tolerances below what we check for with numdiff (and also use a small initial time step).

### CODE REVIEW

**Review by bangerth — APPROVED** [review]

File: tests/sundials/arkode_04.cc [hunk 1]
```diff
@@ -94,26 +95,21 @@ main(int argc, char **argv)
...
-    J       = 0;
-    J(0, 0) = 1;
-    J(1, 1) = 1;
-    J(2, 2) = 1 + gamma / eps;
-    J.gauss_jordan();
+  ode.jacobian_times_vector = [&](const VectorType &src,
+                                  VectorType &      dst,
+                                  double            t,
+                                  const VectorType & /*y*/,
+                                  const VectorType & /*fy*/) -> int {
```
Comment (bangerth) [inline tests/sundials/arkode_04.cc:102]: On second thought, the documentation of this test says that it specifically checks `solve_jacobian_system`, but you are replacing this here with `jacobian_times_vector`.

Comment (drwells) [reply tests/sundials/arkode_04.cc:102]: Yes - `solve_jacobian_system()` was only for sundials < 4.0.0. Is it OK if I address this when I do the big doc update?

Comment (bangerth) [reply tests/sundials/arkode_04.cc:102]: Yes. But could you just update the comment at the top?

**Comment by bangerth** [comment]
It looks like we broke the master branch since one of the testers fails :-( I wonder whether you could rebase on a commit from a couple of days ago to avoid the issue?

File: tests/sundials/arkode_04.cc (hunk 1 above)
Comment (bangerth) [reply tests/sundials/arkode_04.cc:102]: It's ok if you addressed the comment bit in a later PR.

**Review by marcfehling — APPROVED** [review]
Do you need to update the output files? I thought `numdiff` could deal with the changes gracefully.

File: tests/sundials/arkode_04.cc (hunk 1 above)
Comment (drwells) [reply tests/sundials/arkode_04.cc:102]: I need to rebase anyway so I fixed it :)

**Comment by bangerth** [comment]
The test failure should be fixed by #13896, so I'm going to ignore it.

**Pull request merged**